velox/connectors/hive/iceberg/IcebergSplit.cpp

```diff
@@ -30,15 +30,21 @@ HiveIcebergSplit::HiveIcebergSplit(
         _partitionKeys,
     std::optional<int32_t> _tableBucketNumber,
     const std::unordered_map<std::string, std::string>& _customSplitInfo,
-    const std::shared_ptr<std::string>& _extraFileInfo)
+    const std::shared_ptr<std::string>& _extraFileInfo,
+    const std::unordered_map<std::string, std::string>& _infoColumns)
     : HiveConnectorSplit(
           _connectorId,
           _filePath,
           _fileFormat,
           _start,
           _length,
           _partitionKeys,
-          _tableBucketNumber) {
+          _tableBucketNumber,
+          _customSplitInfo,
+          _extraFileInfo,
+          {},
+          0,
+          _infoColumns) {
```

Comment:
> Having both signatures confuses the test calls using default arguments as they don't know which option to pick up.

In the past, we overcame this by adding a new temporary CMake option. The old or new API would be enabled based on this option.

My concern is that  `_infoColumns` will always be set by the clients. We can see that from the `HiveIcebergSplit` in this PR.
 The other parameter defaults have no meaning since they always have to be set.

Reply:
So Prestissimo/Spark runtimes set all the parameters. The defaults are only for tests. And the tests don't need to have infoColumns.

Reply:
The class (and its base) should have been a plain struct in the first place.  I would agree we keep it like this by appending at end for now.  If we want to refactor, change it to a plain struct so we don't have this problem in the future.

Reply:
Thanks @Yuhta. Resolving this comment.